app/lib/logic/contracting.rb

```diff
@@ -45,6 +45,7 @@ def provider_builds_application_for(buyer, application_plan, application_attrs =
         application.validate_human_edition!
 
         application.plan_id = application_plan.id
+        application_attrs.delete 'plan_id' # TODO: is there a better way to do this?
         application.attributes = application_attrs
```

Comment:
if this is a hash, you can do `application.attributes = application_attrs.except('plan_id')`